internal/services/filehash/filesystemhasher.go

```diff
@@ -0,0 +1,107 @@
+package filehash
+
+import (
+	"context"
+	"hash"
+
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/utils/fshasher"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/hasher"
+	"github.com/sirupsen/logrus"
+)
+
+type FileHash struct {
+	Path string
+	Hash string
+}
+
+type FileSystemHasher struct {
+	log     *logrus.Logger
+	alg     string
+	workers int
+}
+
+func NewFileSystemHasher(log *logrus.Logger, alg string, workers int) *FileSystemHasher {
+	return &FileSystemHasher{
+		log:     log,
+		alg:     alg,
+		workers: workers,
+	}
+}
+
+// Calculate calculate file hashes synchronously and store into slice
+func (fhs *FileSystemHasher) CalculateAll(ctx context.Context, path string) ([]FileHash, error) {
+	hashChan := make(chan FileHash)
+	result := make([]FileHash, 0, 1024)
+	var err error
+
+	go func() {
+		hashFuncBuilder := fshasher.FileHasherByHash(func() hash.Hash { return hasher.NewHashSum(fhs.alg) })
```

Comment:
1. The fhs.alg contains just a string that already available by Viper. Can be optimized.
2. the instance of hasher is required only for worker, not here.
3 It's confusing (not only here): path means fullFileName or pathToDir for scanning?

Reply:
1. I prefere read configuration in some init function and pass as params (do not use viper directly in business logic)
2. fshasher launch few jobs so each require own instance of hash.Has, therefore I pass constructor function

Reply:
3. done